pkg/api/validate/types.go

```diff
@@ -35,7 +35,7 @@ var (
 	// This regexp is to guard against InvalidDomainNameLabel for hostname validation
 	rxCloudDomainLabel = regexp.MustCompile(`^[a-z][a-z0-9-]{1,61}[a-z0-9]\.`)
 
-	rxBlobContainerName = regexp.MustCompile(`^[a-z0-9-]{3,63}$`)
+	rxBlobContainerName = regexp.MustCompile(`^[a-zA-Z0-9-]{3,63}$`)
```

Comment:
@ehashman this..

Reply:
Seems fine to me... unless containers are case-insensitive?

Reply:
OK, I see the problem. This is correct for the container, but too restrictive for the blob. We are using container name validation on a blob name. 
https://docs.microsoft.com/en-us/azure/architecture/best-practices/naming-conventions
containers must be lowercase and blobnames can be any "Any URL characters". 

I'll make a new regex..

Reply:
According to the documentation, your implementation of `IsValidBlobName` doesn't take care of some requirements for valid blob names:
- A blob name must be at least one character long and cannot be more than 1,024 characters long, for blobs in Azure Storage.
- Avoid blob names that end with a dot (.), a forward slash (/), or a sequence or combination of the two.

```
=== RUN   TestIsValidBlobName
--- FAIL: TestIsValidBlobName (0.00s)
    types_test.go:347: //bad//: wanted valid false, got true
    types_test.go:347: : wanted valid false, got true
    types_test.go:347: xyypbtodfvqhbjggywtfuboirjpiytlqxpshwoucrwvpturicdagbwyqveglmpgckdjzzxjgyqtjzrqqlwkukdcifloehxffiwtpufdpgkrvyxbjmvipcmzdmprjoeiinsjkuyimtykptvghdwaaqaonopsmrtdfyhtiaqevxvdkdwtzdxefesbpjdrffopbmsinzcsvuxqkxykneqqrxavsywpisddgvzjswkqpwcnqpzogbcuxwvrcqikuzecurigjboopqaxsdzidxmlroblfybhhajwxffjhkykqgxagwcyyegesjxmubyyoaotkypnisujojmjwycshwhfyyefjgpraqgbqizwjbnpivaiutkptfyeynwsnpeybpllvqxjjcqwvfzdmmrhkvqzkafaakshgwbeewxbnytidqvcuuwnxxqnfmypsqjsengwiziwyjmdygaljavdozqpgauyimdpxtgfvjfakmzwxnysuffnvzlpwlhcxajinaasoxtsbejwbgcnyrugdakftvlriiugvlktgohmbhxdguellidywqtwnhvseirxffqnyfemafqdvuenlhrwzanmynwupsclybzhujdwlarydxyjanomttpnlasbphvqyvwtnlscjlgqrjknzqvlifqsjfqvdxelwmihbarzhzlrdevemufrmfkziuzhtvffucohekspkkkajzpwfdnsokcdpffbsdpykjvpcgmfbnaiiruwlbopveomhqzacokwthirhiehjdsufmdmesehygjboxbsgkblroeyztjxdjgnrqevuhvttpbbjxhylgnhczrmcrtekahnfruhgjmobwvnexipotpceqlaypptoolhinevldsagxytzufetiqueibuzivlholrmdhxmsseskfsxwxyiatgvapnlmthnrjbgicisafksdhbpnjqeiktxyskmdawoieuhamnyxnsnpzcbwhkouklhcfiocyhxurmgaboqdzvwmpvooadvrapkjjayt: wanted valid false, got true
    types_test.go:347: bad.: wanted valid false, got true
FAIL

```